stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/GiphyMediaAttachmentView.kt

```diff
@@ -40,18 +50,65 @@ internal class GiphyMediaAttachmentView : ConstraintLayout {
     }
 
     private fun init(attrs: AttributeSet?) {
-        style = MediaAttachmentViewStyle(context, attrs)
+        style = GiphyMediaAttachmentViewStyle(context, attrs)
         binding.loadingProgressBar.indeterminateDrawable = style.progressIcon
         binding.giphyLabel.setImageDrawable(style.giphyIcon)
+        binding.imageView.scaleType = style.scaleType
+        binding.imageView.setBackgroundColor(style.imageBackgroundColor)
     }
 
-    fun showGiphy(attachment: Attachment) {
-        val url = attachment.giphyUrl(GiphyInfoType.FIXED_HEIGHT) ?: attachment.let {
+    /**
+     * Displays a Giphy inside its container. Depending on [GiphyMediaAttachmentViewStyle.giphyConstantSizeEnabled]
+     * it displays the Giphy either inside of a constant size or a resizeable container.
+     */
+    public fun showGiphy(attachment: Attachment) {
+        val url = attachment.giphyUrl(style.giphyType) ?: attachment.let {
             it.imagePreviewUrl ?: it.titleLink ?: it.ogUrl
         } ?: return
 
+        if (style.giphyConstantSizeEnabled) {
+            showConstantSizeGiphy(url)
+        } else {
+            showResizeableGiphy(url)
+        }
+    }
+
+    /**
+     * Displays the Giphy image inside of a constant size container.
+     */
+    private fun showConstantSizeGiphy(url: String) {
+        binding.mediaAttachmentContent.updateLayoutParams {
+            this.height = style.giphyMaxHeight
+            this.width = style.giphyMaxHeight
+        }
+
         binding.loadImage.updateLayoutParams {
-            this.height = GIPHY_INFO_DEFAULT_HEIGHT_DP.dpToPx()
+            this.height = style.giphyMaxHeight
+            this.width = style.giphyMaxHeight
+        }
+
+        CoroutineScope(DispatcherProvider.Main).launch {
+            binding.imageView.setImageDrawable(style.placeholderIcon)
+
+            binding.imageView.load(
+                data = url,
+                placeholderDrawable = style.placeholderIcon,
+                onStart = { binding.loadImage.isVisible = true },
+                onComplete = { binding.loadImage.isVisible = false }
+            )
+        }
+    }
+
+    /**
+     * Displays the Giphy image inside of a resizeable container.
+     */
+    private fun showResizeableGiphy(url: String) {
+        binding.mediaAttachmentContent.updateLayoutParams {
+            this.height = style.giphyMaxHeight
```

Comment:
Updating layour params both on the actual `ShapeableImageView` and the container allows us to dynamically resize the container without having container sizes shifting during or after load.

The only edgecase noticed is when we have an image that is smaller than the actual container size. 

i.e. try setting this in styles:

```xml
<item name="streamUiGiphyMediaAttachmentGiphyMaxHeight">250dp</item>
<item name="streamUiGiphyMediaAttachmentGiphyType">fixedHeight</item>
```

You will see the container "jump" after loading.
Where as this:

```xml
<item name="streamUiGiphyMediaAttachmentGiphyMaxHeight">250dp</item>
<item name="streamUiGiphyMediaAttachmentGiphyType">original</item>
```

Will not result in it jumping since the loaded size is larger than the container's pre loaded size.